modules/redpanda/options.go

```diff
@@ -170,3 +173,12 @@ func WithBootstrapConfig(cfg string, val any) Option {
 		o.ExtraBootstrapConfig[cfg] = val
 	}
 }
+
+// WithEnableAdminAPIAuthentication enables Admin API Authentication.
+// It sets `admin_api_require_auth` configuration to true and configures a bootstrap user account.
+// See https://docs.redpanda.com/current/deploy/deployment-option/self-hosted/manual/production/production-deployment/#bootstrap-a-user-account
+func WithEnableAdminAPIAuthentication() Option {
```

Comment:
suggestion, not a blocker: we have a mix of `WithEnableX` and `WithX` functional options. I'd prefer the latter, but it's a personal choice.

```suggestion
func WithAdminAPIAuthentication() Option {
```

Reply:
Renamed the function to your suggestion.